charts/cloudzero-agent/src/validate.py

```diff
@@ -25,6 +27,45 @@ def check_service_availability(url: str, key: str) -> Tuple[str, str]:
     return (key, "failure")
 
 
+def get_api_key():
+    secret_file_path = "/etc/config/prometheus/secrets/value"
+
+    try:
+        with open(secret_file_path, "r") as file:
+            secret_value = file.read().strip()
+        return secret_value
+    except FileNotFoundError:
+        print(f"Secret file {secret_file_path} not found.", file=sys.stderr)
+        return None
+
+
+def check_external_validation_endpoint() -> Tuple[str, str]:
+    host: str = os.getenv("CZ_HOST", "api.cloudzero.com")
+    api_key = get_api_key()
+
+    # Use for testing
+    # host = "dev-api.cloudzero.com"
```

Comment:
How should we handle this for testing? When running locally, should we have a flag for overriding host?

Reply:
Good question - I have been adding elements to the `values.yaml` which allow me to set.